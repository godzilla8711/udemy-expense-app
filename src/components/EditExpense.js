import React from 'react';

class EditExpense extends React.Component {
  render() {
    return (
      <div>
        Edit the details of Expense ID
      </div>
    );
  }
}

export default EditExpense;
