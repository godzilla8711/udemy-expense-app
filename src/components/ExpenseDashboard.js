import React from 'react';
import ReactDOM from 'react-dom';

class ExpenseDashboard extends React.Component {
  render() {
    return (
      <div>
        This is the Expense Dashboard Component.
      </div>
    );
  }
}

export default ExpenseDashboard;
