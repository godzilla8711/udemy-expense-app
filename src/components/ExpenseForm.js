import React from 'react';

class ExpenseForm extends React.Component {
  constructor(props) {
    super(props);
    this.onChangeDescription = this.onChangeDescription.bind(this);
    this.onChangeAmount = this.onChangeAmount.bind(this);
    this.onChangeNote = this.onChangeNote.bind(this);

    // Set the default state.
    this.state = {
      description: '',
      amount: '',
      note: ''
    };
  }

  componentDidMount() {
    // Set default focus.
    this.descriptionInput.focus();
  }

  onChangeDescription(e) {
    const description = e.target.value;
    this.setState(() => ({
      description
    }));
  }

  onChangeAmount(e) {
    const amount = e.target.value;
    this.setState(() => ({
      amount
    }));
  }

  onChangeNote(e) {
    const note = e.target.value;
    this.setState(() => ({
      note
    }));
  }

  render() {
    return (
      <div>
        <h2>Expense Form</h2>
        <input
          type="text"
          placeholder="Description"
          value={this.state.description}
          onChange={this.onChangeDescription}
          ref={thisElement => (this.descriptionInput = thisElement)}
        />
        <input
          type="text"
          placeholder="Amount"
          value={this.state.amount}
          onChange={this.onChangeAmount}
        />
        <textarea
          placeholder="Note"
          value={this.state.note}
          onChange={this.onChangeNote}
        />
      </div>
    );
  }
}

export default ExpenseForm;
