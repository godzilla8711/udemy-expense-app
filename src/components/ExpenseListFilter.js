import React from 'react';
import { connect } from 'react-redux';

import { filterText, sortByType } from '../actions/filter-action';

class ExpenseListFilter extends React.Component {
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text} onChange={e => {
            this.props.dispatch(filterText(e.target.value));
          }}
        />
        <select value={this.props.filters.sortBy} onChange={e => {
          const selectedValue = e.target.value;
          if (selectedValue) {
            this.props.dispatch(sortByType(selectedValue));
          }
          }}
        >
          <option value="amount">Amount</option>
          <option value="date">Date</option>
        </select>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    filters: state.filters
  };
}

export default connect(mapStateToProps)(ExpenseListFilter);
