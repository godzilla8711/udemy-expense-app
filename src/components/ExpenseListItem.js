import React from 'react';
import { connect } from 'react-redux';
import { removeExpense } from '../actions/expense-action';

class ExpenseListItem extends React.Component {
  render() {
    const { expense } = this.props;
    return (
      <div>
        <h3>Description: {expense.description}</h3>
        <p>Amount: {expense.amount}   Created Date: {expense.createdAt}</p>
        <button onClick={e => this.props.dispatch(removeExpense(expense.expenseId))} >
          Remove Expense
        </button>
      </div>
    );
  }
}

export default connect()(ExpenseListItem);
