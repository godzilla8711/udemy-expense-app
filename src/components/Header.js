import React from 'react';
import { NavLink } from 'react-router-dom';

class Header extends React.Component {
  render() {
    return (
      <div>
        <h2>Expense App</h2>
        <NavLink to="/" activeClassName="is-active" exact={true}>Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add Evidence</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Edit Evidence</NavLink>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
      </div>
    );
  }
}

export default Header;
