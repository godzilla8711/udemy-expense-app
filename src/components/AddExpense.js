import React from 'react';

import ExpenseForm from './ExpenseForm';

class AddExpense extends React.Component {
  render() {
    return (
      <div>
        <h2>Add a new expense</h2>
        <ExpenseForm />
      </div>
    );
  }
}

export default AddExpense;
