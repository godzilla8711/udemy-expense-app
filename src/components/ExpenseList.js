import React from 'react';
import { connect } from 'react-redux';

import ExpenseListItem from './ExpenseListItem';
import getVisibleExpenses from '../selectors/expense-selector';

export class ExpenseList extends React.Component {
  render() {
    return (
      <div>
        <h2>List of {this.props.expenses.length} expenses</h2>
        {this.props.expenses.map(expense => {
          return <ExpenseListItem key={expense.description} expense={expense} />;
        })}
      </div>
    );
  }
}

export function mapStateToProps(state) {
  return {
    expenses: getVisibleExpenses(state.expenses, state.filters)
  };
}

export default connect(mapStateToProps)(ExpenseList);
