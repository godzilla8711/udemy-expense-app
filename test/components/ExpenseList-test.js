import React from 'react';
import { shallow } from 'enzyme';

import { ExpenseList } from '../../src/components/ExpenseList';
import expenseItems from '../mock-data/expense-data';

describe('render', () => {
  it('renders component with items', () => {
    const wrapper = shallow(<ExpenseList expenses={expenseItems} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders component with empty items', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
