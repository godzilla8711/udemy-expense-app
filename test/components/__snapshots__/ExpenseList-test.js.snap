// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders component with empty items 1`] = `
<div>
  <h2>
    List of 
    0
     expenses
  </h2>
</div>
`;

exports[`render renders component with items 1`] = `
<div>
  <h2>
    List of 
    3
     expenses
  </h2>
  <Connect(ExpenseListItem)
    expense={
      Object {
        "amount": 54500,
        "createdOn": 1514955600000,
        "description": "January Rent",
        "expenseId": 1001,
        "note": "Final rent check for January",
      }
    }
    key="January Rent"
  />
  <Connect(ExpenseListItem)
    expense={
      Object {
        "amount": 72500,
        "createdOn": 1525665600000,
        "description": "May Rent",
        "expenseId": 1002,
        "note": "Final rent check for May",
      }
    }
    key="May Rent"
  />
  <Connect(ExpenseListItem)
    expense={
      Object {
        "amount": 38500,
        "createdOn": 1520226000000,
        "description": "February Rent",
        "expenseId": 1003,
        "note": "Final rent check for February",
      }
    }
    key="February Rent"
  />
</div>
`;
